lib/puppet/type/file/content.rb

```diff
@@ -119,6 +120,20 @@ def insync?(is)
       result
     end
 
+    def property_matches?(current, desired)
+      basic = super
+      # The inherited equality is always accepted, so use it if valid.
+      time_types = ["mtime", "ctime"]
+      return basic if basic || !time_types.include?(checksum_type.to_s)
+      return false unless current && desired
+      begin
+        raise "not a time checksum" if !time_types.include?(sumtype(current)) || !time_types.include?(sumtype(desired))
+        DateTime.parse(current) >= DateTime.parse(desired)
```

Comment:
self.fail instead of raising an error?

Yeah, just sumdata makes sense, parse is a bit rogue.